e2etest/bgp_test.go

```diff
@@ -390,6 +390,167 @@ var _ = ginkgo.Describe("BGP", func() {
 			}),
 		)
 	})
+
+	ginkgo.Context("validate configuration changes", func() {
+		ginkgo.AfterEach(func() {
+			// Clean previous configuration.
+			err := updateConfigMap(cs, configFile{})
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("request a specific address pool for each Loadbalancer service", func() {
+			ginkgo.By("configure first addresspool")
```

Comment:
Can we do the two iterations inside a loop?
That'd remove some code duplication